### PR TITLE
Create and use style variable for background color - RSC-447

### PR DESCRIPTION
**Pull request by rpokorny**
https://issues.sonatype.org/browse/RSC-447

Stemming from @glenhunter 's comment here https://github.com/sonatype/sonatype-react-shared-components/pull/271#issuecomment-758913952 and @DarthHater 's recent work, we are starting to think about making it easier to create an alternative color theme for RSC, and/or allowing consuming projects to easily override colors used within RSC.  Towards that end, this PR adds a CSS custom property for the background color of typical components.  Being a CSS custom prop allows it to be easily overridden in consuming projects.

### CODE REVIEW

File: lib/src/base-styles/_nx-btn.scss [hunk 1]
```diff
@@ -21,7 +21,7 @@
   @include container-horizontal;
   @extend %nx-btn--disabled;
 
-  background-color: $nx-white;
+  background-color: var(--rsc-component-background-color);
```
Comment (glenhunter) [inline lib/src/base-styles/_nx-btn.scss:24]: I don't think that we want to use this variable on buttons, we'd want a separate set for the various button types

File: lib/src/base-styles/_nx-form-select.scss [hunk 2]
```diff
@@ -11,7 +11,7 @@
   -webkit-appearance: none;
   -moz-appearance: none;
   appearance: none;
-  background: #fff url("data:image/svg+xml;utf8, \
+  background: var(--rsc-component-background-color) url("data:image/svg+xml;utf8, \
```
Comment (glenhunter) [inline lib/src/base-styles/_nx-form-select.scss:14]: I'm not sure that we would want this, the text input, or the policy slider to also be tied to this variable. I think a form element background colour variable might be a better way to go.

File: lib/src/base-styles/_nx-btn.scss (hunk 1 above)
Comment (rpokorny) [reply lib/src/base-styles/_nx-btn.scss]: > I don't think that we want to use this variable on buttons, we'd want a separate set for the various button types

aa490bf3

File: lib/src/base-styles/_nx-form-select.scss (hunk 2 above)
Comment (rpokorny) [reply lib/src/base-styles/_nx-form-select.scss]: > I'm not sure that we would want this, the text input, or the policy slider to also be tied to this variable. I think a form element background colour variable might be a better way to go.

aa490bf3

**Comment by rpokorny** [comment]
Renamed the variable to use `nx` as the prefix rather than `rsc`.  For consistency with our other variables. 96696627

**Review by glenhunter — APPROVED** [review]

**Pull request merged**